### PR TITLE
Listen for notification when Insulin Delivery Store updates its cache

### PR DESCRIPTION
**Pull request by aranasaurus**
This addresses #1037 and requires changes in https://github.com/LoopKit/LoopKit/pull/304

### CODE REVIEW

File: Loop/Managers/LoopDataManager.swift [hunk 1]
```diff
@@ -126,6 +126,19 @@ final class LoopDataManager {
...
+            },
+            NotificationCenter.default.addObserver(
+                forName: .InsulinDeliveryStoreCacheDidChange,
+                object: doseStore.insulinDeliveryStore,
+                queue: OperationQueue.main
+            ) { (note) in
+                self.dataAccessQueue.async {
+                    self.logger.default("Received notification of cached dosing changing")
+
+                    self.insulinEffect = nil
```
Comment (aranasaurus) [inline Loop/Managers/LoopDataManager.swift]: I wasn't really sure about this line, but it's what the other bolus change update was doing so I figured it was probably necessary 🤷‍♂

Comment (ps2) [reply Loop/Managers/LoopDataManager.swift]: This is necessary to trigger a recalc of insulin effect.

Comment (aranasaurus) [reply Loop/Managers/LoopDataManager.swift]: Thanks for the clarification!

File: Loop/Managers/LoopDataManager.swift [hunk 2]
```diff
@@ -126,6 +126,19 @@ final class LoopDataManager {
...
+                    self.notify(forChange: .bolus)
+                }
+            },
+            NotificationCenter.default.addObserver(
+                forName: .InsulinDeliveryStoreCacheDidChange,
+                object: doseStore.insulinDeliveryStore,
+                queue: OperationQueue.main
+            ) { (note) in
+                self.dataAccessQueue.async {
+                    self.logger.default("Received notification of cached dosing changing")
```
[the remark this answers is not in the document]
Comment (aranasaurus) [reply Loop/Managers/LoopDataManager.swift]: ✅ https://github.com/LoopKit/Loop/pull/1150/commits/11a1a03ba5067a4d561fb5106ecec822e3c1d882

**Comment by ps2** [comment]
Doing a little cleanup on names:

https://github.com/LoopKit/LoopKit/pull/307
https://github.com/LoopKit/Loop/pull/1165

File: Loop/Managers/LoopDataManager.swift [hunk 3]
```diff
@@ -126,6 +126,19 @@ final class LoopDataManager {
 
                     self.insulinEffect = nil
 
+                    self.notify(forChange: .bolus)
+                }
+            },
+            NotificationCenter.default.addObserver(
```
Comment (ps2) [inline Loop/Managers/LoopDataManager.swift]: The `addObserver` above this is not specifying a name for notifications from doseStore. Could you add the right name? I believe it is DoseStore.valuesDidChange.

**Comment by ps2** [comment]
This can be merged if it is rebased and updated with the new notification name.

File: Loop/Managers/LoopDataManager.swift (hunk 3 above)
Comment (aranasaurus) [reply Loop/Managers/LoopDataManager.swift]: Yep I’ll probably be able to do that tomorrow night or this weekend.

**Pull request closed without merging**